### ISSUE TITLE
Command line arguments

### ISSUE THREAD

**Issue by litchfld** [alcap-org/AlcapDAQ#155]
**TL;DR**:  I'm planning to change the meaning of `arguments.start` and `arguments.stop` (but not `main::start` and `main::stop`). Is anyone relying on the current meaning?

**Full explanation**: When parsing the arguments list of rootana we have the option -n:

```
  -n <count>             Analyze only <count> events.
  -n <first> <last>      Analyze only events from <first> to <last>.
```

Which is handled by this nightmare-looking option case:

``` C++
   case 'n':
     if(i+1 < argc){
       if(isNumber(argv[i+1])){
         arguments.start = atoi(argv[i+1]);
         i+=2;
       }
       else {
         std::cerr << "ERROR: Argument " << argv[i+1]
                   << " for option -n is not a number\n";
         help_command_line(argv[0]); return 1;
       }
       if(i < argc && argv[i][0] != '-'){
         if(isNumber(argv[i])){
           arguments.stop = atoi(argv[i]);
           i+=1;
         }
         else{
           std::cerr << "ERROR: Argument " << argv[i+1]
                     << " for option -n is not a number\n";
           help_command_line(argv[0]); return 1;
         }
       }
     }
     else{
       std::cerr << "ERROR: No argument for number of processed events specified\n";
       help_command_line(argv[0]); return 1;
     }
     break;
```

This process is then inverted in a reverse nightmare in  main.cpp:

``` C++
  Long64_t start = 0;
  Long64_t stop = nentries;
  if(arguments.stop>0){
    if(arguments.start>0)
      start = (Long64_t)(arguments.start-1);
    if((Long64_t)arguments.stop < nentries)
      stop = (Long64_t)arguments.stop;
  }
  else if((Long64_t)arguments.start < nentries && arguments.start > 0){
    stop = (Long64_t)arguments.start;
  }
  gTotalEntries=&stop;
```

If I have parsed all that correctly the logic is thus:

_In CommandLine.cpp_
- IF (you find an n):
  - set `arguments.start` to the number after 'n'
  - IF (there is another number):
    - set `arguments.stop` to the 2nd number after 'n'
  - ELSE 
    - set `arguments.stop` to a null value

_In main.cpp_
- IF (`arguments.stop` is not a null value):
  - start at MAX [`arguments.start` -1, 0]
  - stop at MIN [`arguments.stop`, N_entries]
- ELSE IF `arguments.start` is not outside the valid range
  - set `arguments.stop` to `arguments.start`

This could all be made much more logical and transparent (and therefore easier for me to replace in the navigator) if instead we just do:
_In ComandLine.h_
- IF (there are two numbers after 'n'):
  - Use the first one for `arguments.start`, and the second one for `arguments.stop`
- ELSE IF (there is one number)
  - Take `arguments.start` to be zero, and `arguments.stop` from the number

_in main.cpp_
- start at `arguments.start`
- stop at `arguments.stop` - 1 

This does have a side effect though, anything else who has access to the arguments struct sees the meaning of `arguments.start` and `arguments.stop` switched if only one number is passed in. To be precise they see the values corresponding to the normal meaning of "start" and "stop" rather than the inverted meanings they have now.  Which is (IMNSHO) more sane, but would break code designed to cope with the current situation.

Anyone know of any code that relies on the current behaviour, or any other reason not to do this?

**Comment by jrquirk** [comment]
I've never used either of these variables, and have no objections otherwise either.

**Comment by benkrikler** [comment]
I don't think they're global variables anyway and only local within some
functions of main.cpp (i think).

The one part that is related and currently used in several modules is the
global variable `gTotalEntries` which is currently assigned to the value of
arguments.stop in the `Main_event_loop` function.  This shouldn't be hard
to reassign in your new approach though.  That variable was only a
quick-fix anyway under the assumption that the Event or Setup navigators
would shortly take over this functionality.

**Comment by litchfld** [comment]
> The one part that is related and currently used in several modules is the global variable `gTotalEntries`...

gTotalEntries is no problem, it gets set after all of the juggling has finished.  

So i'll consider that okayed.
